frontend/src/views/components/DrawerContents.tsx

```diff
@@ -14,27 +14,30 @@ import { MainButton } from "./MainButton";
 import styled from "styled-components";
 import { useDispatch, useSelector } from "react-redux";
 import { selectTheme } from "../../reducer/colorModeSlice";
-import { logout, selectUser } from "../../reducer/userSlice";
-import { AppDispatch } from "../../store/store";
 import { useState } from "react";
+import { logout } from "../../reducer/userSlice";
+import { AppDispatch } from "../../store/store";
+import { UserService } from "../../services/users/service";
 
 export const DrawerContents = () => {
+  const dispatch: AppDispatch = useDispatch();
   const navigate = useNavigate();
   const theme = useSelector(selectTheme);
-  const account = useSelector(selectUser);
-  const dispatch: AppDispatch = useDispatch();
   const [selectedId, setSelectedId] = useState(0);
 
   const navigatePage = (link: string, num: number) => {
     navigate(link);
     setSelectedId(num);
   };
 
-  const handleLogput = () => {
-    if (account.isLogin) {
+  const handleLogout = async () => {
+    setSelectedId(4);
```

Comment:
This is also okay, but I couldn't understand what id 4 means. To make it readable, you may need to assign constants.

Reply:
I also couldn't understand why id is used here. It seems unnecessary so I removed it and 
I made a constant file for toolbarItems. According to this, I modified the code like below.
```js

 const handleLogout = async (path: string) => {
    const { isError, message } = await UserService.signOut();
    if (isError) {
      console.error("Failed to logout: ", message);
    } else {
      dispatch(logout());
      navigate(path);
    }
  };

  const handleClick = (item: ToolbarItem) => {
    item.name === "Logout" ? handleLogout(item.path) : navigate(item.path);
  };

      <StyledList>
        {toolbarItems.map((item, index) => (
          <ListItem key={index} disablePadding>
            <StyledListItemButton
              onClick={() => handleClick(item)}
              disableRipple
            >
              <ListItemIcon>
                <item.icon style={{ color: theme.palette.secondary.main }} />
              </ListItemIcon>
              <ListItemText primary={item.name} />
            </StyledListItemButton>
          </ListItem>
        ))}
      </StyledList>
```